bindings/rascal/utils/cur.py

```diff
@@ -119,86 +142,96 @@ def fit_transform(self, managers):
-                    if (idx >= strides_by_sp[sp][i_manager[sp]] and
-                                    idx < strides_by_sp[sp][i_manager[sp] + 1]):
-                        selected_ids[i_manager[sp]].append(
-                            map_by_manager[sp][i_manager[sp]][idx])
-                        carry_on = False
-                    else:
-                        i_manager[sp] += 1
-        for ii in range(len(selected_ids)):
-            selected_ids[ii] = list(np.sort(selected_ids[ii]))
-        return selected_ids
+        if self.act_on == 'sample per species':
+            sps = list(n_select.keys())
+            # get various info from the structures about the center atom species and indexing
+            (strides_by_sp, global_counter, map_by_manager,
+             indices_by_sp) = get_index_mappings_sample_per_species(managers, sps)
+            selected_ids_by_sp = {key: np.sort(val[:n_select[key]])
+                                  for key, val in self.selected_sample_ids_by_sp.items()}
+            self.selected_ids = convert_selected_global_index2perstructure_index_per_species(
+                managers, selected_ids_by_sp, strides_by_sp, map_by_manager, sps)
+            # return self.selected_ids
+            # build the pseudo points
+            pseudo_points = SparsePoints(self._representation)
+            pseudo_points.extend(managers, self.selected_ids)
+            return pseudo_points
+
+        elif self.act_on == 'sample':
+            selected_ids_global = np.sort(self.selected_sample_ids[:n_select])
+            strides, _, map_by_manager = get_index_mappings_sample(managers)
+            self.selected_ids = convert_selected_global_index2perstructure_index(managers,
+                                                                            selected_ids_global, strides, map_by_manager)
+            return self.selected_ids
+            # # build the pseudo points
+            # pseudo_points = SparsePoints(self._representation)
+            # pseudo_points.extend(managers, self.selected_ids)
+            # return pseudo_points
+
+        elif self.act_on == 'feature':
+            feat_idx2coeff_idx = self._representation.get_feature_index_mapping(
+                managers)
+            self.selected_ids = {key: []
+                                 for key in feat_idx2coeff_idx[0].keys()}
+            selected_ids_sorting = np.argsort(
+                self.selected_feature_ids_global[:n_select])
+            selected_feature_ids = self.selected_feature_ids_global[selected_ids_sorting]
+            for idx in selected_feature_ids:
+                coef_idx = feat_idx2coeff_idx[idx]
+                for key in self.selected_ids.keys():
+                    self.selected_ids[key].append(int(coef_idx[key]))
+            # keep the global indices and ordering for ease of use
+            self.selected_ids['selected_features_global_ids'] = selected_feature_ids.tolist(
+            )
```

Comment:
...close-paren on the next line? Was this something `autopep8` did to try to reduce line length? It's over 80 characters in any case, and this certainly doesn't help readability.

Reply:
that's an `autopep8` thing. If I revert this then it will change it back next time I run `make pretty-python` so I would like not the change these kind of things.

Reply:
Ok, all the more reason to switch to a different formatter then, this one just isn't working that well.

Reply:
Linking to #259 for another example of the crap we're getting from that issue

Reply:
I think so too. Here are some [suggestions](https://www.kevinpeters.net/auto-formatters-for-python), it seems to me that `black` is simple and effective to use.

Reply:
but this change has to be done in a separate PR since it really formats differently.